File: ObjectAvoidance/java/src/main/java/elite/math/trig/Arctangent.java
package elite.math.trig;

import elite.math.trig.triangle.Adjacent;
import elite.math.trig.triangle.Opposite;

/**
 *
 * @author CFerg (Elite)
 */
public class Arctangent {

    /**
     *
     */
    public double arctangent;
    
    /**
     *
     * @param a
     * @param o
     */
    public Arctangent(Adjacent a, Opposite o){
        this.arctangent = (a.adjacent / o.opposite);
    }
    
    /**
     *
     * @param tan
     */
    public Arctangent(Tangent tan){
        this.arctangent = (1 / tan.tangent);
    }
    
    /**
     *
     * @param cot
     */
    public Arctangent(Cotangent cot){
        this.arctangent = cot.cotangent;
    }
}

File: ObjectAvoidance/src/elite/math/trig/Arctangent.kt
package elite.math.trig

import elite.math.trig.triangle.Adjacent
import elite.math.trig.triangle.Opposite

/**
 *
 * @author CFerg (Elite)
 */
class Arctangent {

    /**
     *
     */
    var arctangent: Double = 0.toDouble()

    /**
     *
     * @param a
     * @param o
     */
    constructor(a: Adjacent, o: Opposite) {
        this.arctangent = a.adjacent / o.opposite
    }

    /**
     *
     * @param tan
     */
    constructor(tan: Tangent) {
        this.arctangent = 1 / tan.tangent
    }

    /**
     *
     * @param cot
     */
    constructor(cot: Cotangent) {
        this.arctangent = cot.cotangent
    }
}